frontend/src/concepts/projects/ProjectSelectorNavigator.tsx

```diff
@@ -13,15 +13,13 @@ const ProjectSelectorNavigator: React.FC<ProjectSelectorProps> = ({
 }) => {
   const navigate = useNavigate();
   const { namespace } = useParams();
-  const { updatePreferredProject } = React.useContext(ProjectsContext);
+  const { projects, updatePreferredProject } = React.useContext(ProjectsContext);
 
   return (
     <ProjectSelector
       {...projectSelectorProps}
       onSelection={(projectName) => {
-        if (!projectName) {
-          updatePreferredProject(null);
-        }
+        updatePreferredProject(projects.find(byName(projectName)) || null);
```

Comment:
```suggestion
        updatePreferredProject(projectName ? projects.find(byName(projectName)) : null);
```
I favour this ever so slightly -- Projects are sorta not well managed and looping over 500 projects looking for `''` seems unnecessary -- so lets just check if it's a value before looking.